src/test/java/core/basesyntax/service/impl/DataConverterImplTest.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitConverter;
+import core.basesyntax.service.exceptions.InvalidDataException;
+import java.util.List;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class DataConverterImplTest {
+    private DataConverterImpl dataConverter = new DataConverterImpl();
+
+    @Test
+    void convertToTransaction_incorrectData_notOk() {
+        List<String> inputReport = List.of("a,b,c","d,e");
+        Assertions.assertThrows(InvalidDataException.class, ()
+                -> dataConverter.convertToTransaction(inputReport));
+    }
+
+    @Test
+    void convertToTransaction_correctData_Ok() {
+        List<String> inputData = List.of("b,banana,20");
+        List<FruitTransaction> expected = List.of(
+                new FruitTransaction(FruitTransaction.Operation.BALANCE,
+                FruitConverter.convertToFruit("banana"),20));
+        List<FruitTransaction> actual = dataConverter.convertToTransaction(inputData);
+
+        Assertions.assertEquals(expected.toString(),actual.toString());

```

Comment:
It is not recommended to compare the toString() representation of objects for equality checks. It would be better to compare the actual objects using assertEquals.

Reply:
ok, done